store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -1094,34 +1090,15 @@ private boolean checkInDiskByCommitOffset(long offsetPy, long maxOffsetPy) {
     }
 
     private boolean isTheBatchFull(int sizePy, int maxMsgNums, int bufferTotal, int messageTotal, boolean isInDisk) {
-
-        if (0 == bufferTotal || 0 == messageTotal) {
-            return false;
-        }
-
-        if (maxMsgNums <= messageTotal) {
-            return true;
-        }
-
-        if (isInDisk) {
-            if ((bufferTotal + sizePy) > this.messageStoreConfig.getMaxTransferBytesOnMessageInDisk()) {
-                return true;
-            }
-
-            if (messageTotal > this.messageStoreConfig.getMaxTransferCountOnMessageInDisk() - 1) {
-                return true;
-            }
-        } else {
-            if ((bufferTotal + sizePy) > this.messageStoreConfig.getMaxTransferBytesOnMessageInMemory()) {
-                return true;
-            }
-
-            if (messageTotal > this.messageStoreConfig.getMaxTransferCountOnMessageInMemory() - 1) {
-                return true;
-            }
-        }
-
-        return false;
+        return    (0 != bufferTotal && 0 != messageTotal)
```

Comment:
This one looks too complex to understand...I suggest to keep the original look-and-feel.